src/lib.rs

```diff
@@ -166,7 +166,7 @@ impl codec::Encode for Encoded {
 }
 
 /// A phase of a block's execution.
-#[derive(Clone, Debug, Eq, PartialEq, Decode)]
+#[derive(Clone, Debug, Eq, PartialEq, Decode, Encode)]
```

Comment:
was this required `compact encoding/decoding` to work or a bonus? :)

Reply:
Just for the tests in order to encode some `EventRecord`s